src/cmap/connection.ts

```diff
@@ -441,10 +449,16 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
     const message = this.prepareCommand(ns.db, command, options);
 
     let started = 0;
-    if (this.monitorCommands) {
+    if (
+      this.monitorCommands ||
+      (this.description.hello && this.mongoLogger?.willLog(SeverityLevel.DEBUG, this.component))
```

Comment:
There is quite a bit of non-booleans here and it would be nice to be clear about the expectations, so rather than depending on the falsey nature of hello can you check for null, and same goes for the `emitAndLogCommand` function, instead of passing in the hello document, can it only take in a boolean "isBefore/AfterHandshake"

Reply:
sounds good, we talked about this, but just for an added paper trail: checking the hello document exists is not enough truthiness for knowing the handshake has been completed. I introduced a `established` property to the `Connection` class, that is now used in `emitAndLogCommand`.